src/java/org/apache/cassandra/index/sai/utils/WrappedScoredPrimaryKeyIterator.java

```diff
@@ -20,41 +20,18 @@
 
 import java.util.Iterator;
 
-import org.apache.cassandra.utils.Pair;
-
-/**
- * A {@link ScoredRowId} iterator that returns results in score order.
- */
-public interface ScoredRowIdIterator extends Iterator<ScoredRowId>, AutoCloseable
+public class WrappedScoredPrimaryKeyIterator extends ScoredPrimaryKeyIterator
```

Comment:
I would replace this with a CloseableIterator.wrap method

Reply:
I like that, this felt awkward, but I wasn't sure what to do exactly.